enzyme/Enzyme/Runtimes/FPRT/GarbageCollection.cpp

```diff
@@ -0,0 +1,127 @@
+//===- Trace.cpp - FLOP Garbage collection wrappers
+//---------------------------------===//
+//
+//                             Enzyme Project
+//
+// Part of the Enzyme Project, under the Apache License v2.0 with LLVM
+// Exceptions. See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// If using this code in an academic setting, please cite the following:
+// @incollection{enzymeNeurips,
+// title = {Instead of Rewriting Foreign Code for Machine Learning,
+//          Automatically Synthesize Fast Gradients},
+// author = {Moses, William S. and Churavy, Valentin},
+// booktitle = {Advances in Neural Information Processing Systems 33},
+// year = {2020},
+// note = {To appear in},
+// }
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains infrastructure for flop tracing
+//
+// It is implemented as a .cpp file and not as a header becaues we want to use
+// C++ features and still be able to use it in C code.
+//
+//===----------------------------------------------------------------------===//
+
+#include <algorithm>
+#include <array>
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <list>
+#include <llvm/Support/ErrorHandling.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#define ENZYME_ENABLE_GARBAGE_COLLECTION
```

Comment:
not necessary pre pldi, but llvm / enzyme doesn't need to be built on a system with mpfr at the moment [and thus CI fails]